src/proxy/cache/itemStore.js

```diff
@@ -67,57 +78,64 @@ export default function itemStoreFactory(options) {
+            config.itemTTL = ttl;
+        },
+
         /**
          * Get the item form the given key/id/uri
-         * @param {String} key - something identifier
+         * @param {string} key - something identifier
          * @returns {Promise<Object>} the item
          */
-        get: function get(key) {
-            return getStore().then(function(itemStorage) {
-                return itemStorage.getItem(key);
-            });
+        get(key) {
+            if (!this.has(key)) {
+                return Promise.resolve();
+            }
+            return getStore().then(itemStorage => itemStorage.getItem(key));
         },
 
         /**
          * Check whether the given item is in the store
-         * @param {String} key - something identifier
-         * @returns {Boolean}
+         * @param {string} key - something identifier
+         * @returns {boolean}
          */
-        has: function has(key) {
-            return _.contains(index, key);
+        has(key) {
+            return index.has(key) && !isExpired(key);
         },
 
         /**
          * Add/Set an item into the store, under the given key
-         * @param {String} key - something identifier
-         * @param {Object} item - the item
-         * @returns {Promise<Boolean>} chains
+         * @param {string} key - something identifier
+         * @param {object} item - the item
+         * @returns {Promise<boolean>} chains
          */
-        set: function set(key, item) {
-            var self = this;
-            return getStore().then(function(itemStorage) {
-                return itemStorage.setItem(key, item).then(function(updated) {
+        set(key, item) {
+            return getStore().then(itemStorage => {
+                return itemStorage.setItem(key, item).then(updated => {
                     if (updated) {
-                        if (!_.contains(index, key)) {
-                            index.push(key);
+                        if (!index.has(key)) {
+                            list.push(key);
+                            index.set(key, {
+                                timestamp: Date.now()
+                            });
                         }
 
                         if (config.preload) {
-                            _.defer(function() {
-                                itemPreloader.preload(item);
-                            });
+                            _.defer(() => itemPreloader.preload(item));
                         }
                     }
 
                     //do we reach the limit ? then remove one
-                    if (index.length > 1 && index.length > config.maxSize) {
-                        return self.remove(index[0]).then(function(removed) {
-                            return updated && removed;
-                        });
+                    if (list.length > 1 && list.length > config.maxSize) {
+                        return this.remove(list[0]).then(removed => updated && removed);
```

Comment:
The key will remain in the index (until pruning), is it intended?

Reply:
The remove function is taking care of the index. Nothing is removed outside of this function.

Reply:
Sorry, my bad!